src/cdk/a11y/focus-trap/focus-trap.ts

```diff
@@ -26,6 +26,15 @@ import {take} from 'rxjs/operators';
 import {InteractivityChecker} from '../interactivity-checker/interactivity-checker';
 
 
+/** Options for how to handle focus upon opening the dialog */
+export const enum AutoFocusOption {
```

Comment:
Rather than making this a full enum, I would probably make it a string literal enum:
```ts
export type AutoFocusTarget = 'dialog' | 'first-tabbable' | 'first-heading';
```